xmr-btc/tests/harness/storage.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{anyhow, Context, Result};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::path::Path;
+
+pub struct Database<T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    db: sled::Db,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T> Database<T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    const LAST_STATE_KEY: &'static str = "latest_state";
```

Comment:
As this is expected to always be used as a key for sled, then serialize at instantiation to avoid bug like the one in nectar: [`d82417b` (#3316)](https://github.com/comit-network/comit-rs/pull/3316/commits/d82417b7eea2d9b09f371a7887a1efc61dee340d)

Reply:
```
error[E0277]: the trait bound `conquer_once::raw::Lazy<std::vec::Vec<u8>, conquer_once::with_std::ParkThread>: std::convert::AsRef<[u8]>` is not satisfied
  --> xmr-btc/tests/harness/storage.rs:27:37
   |
27 |         let old_value = self.db.get(&LAST_STATE_KEY)?;
   |                                     ^^^^^^^^^^^^^^^
   |                                     |
   |                                     the trait `std::convert::AsRef<[u8]>` is not implemented for `conquer_once::raw::Lazy<std::vec::Vec<u8>, conquer_once::with_std::ParkThread>`
   |                                     help: consider adding dereference here: `&*LAST_STATE_KEY`
   |
```
Cant get it to compile when i use conquer once. I have put a todo

Reply:
Read the help that the compiler gave you...